### PR TITLE
Spyder v5.0.5 modules for foss/2020b and intel/2020b

### PR DESCRIPTION
**Pull request by SimonPinches**
(created using `eb --new-pr`)

### CODE REVIEW

File: easybuild/easyconfigs/s/Spyder/Spyder-5.0.5-foss-2020b.eb [hunk 1]
```diff
@@ -0,0 +1,278 @@
...
+builddependencies = [
+    ('CMake', '3.18.4'),
+]
+dependencies = [
+    ('Python', '3.8.6'),
+    ('ZeroMQ', '4.3.3'),
+    ('PyQt5', '5.15.1'),
+    ('IPython', '7.18.1'),
+    ('geopandas', '0.9.0'),
+    ('libspatialindex', '1.9.3'),
```
Comment (Micket) [inline easybuild/easyconfigs/s/Spyder/Spyder-5.0.5-foss-2020b.eb:22]: I really don't think spyder needs these at all. Without it, we can drop this all the way down to `GCCcore`.

File: easybuild/easyconfigs/s/Spyder/Spyder-5.0.5-foss-2020b.eb [hunk 2]
```diff
@@ -0,0 +1,278 @@
...
+local_spyder_preinstallopts = (
+    """sed -i "s/'pyqt5.*,//g" setup.py && """
+    """sed -i "s/'pyqtwebengine.*,//g" setup.py && """
+)
+
+exts_list = [
+    ('astroid', '2.6.5', {
+        'checksums': ['83e494b02d75d07d4e347b27c066fd791c0c74fc96c613d1ea3de0c82c48168f'],
+    }),
+    ('async_generator', '1.10', {
```
Comment (Micket) [inline easybuild/easyconfigs/s/Spyder/Spyder-5.0.5-foss-2020b.eb:40]: Already included in deps

File: easybuild/easyconfigs/s/Spyder/Spyder-5.0.5-foss-2020b.eb [hunk 3]
```diff
@@ -0,0 +1,278 @@
...
+        'checksums': ['6ebb3d106c12920aaae42ccb6f787ef5eefdcdd166ea3d628fa8476abe712144'],
+    }),
+    ('autopep8', '1.5.7', {
+        'checksums': ['276ced7e9e3cb22e5d7c14748384a5cf5d9002257c0ed50c0e075b68011bb6d0'],
+    }),
+    ('binaryornot', '0.4.4', {
+        'checksums': ['359501dfc9d40632edc9fac890e19542db1a287bbcfa58175b66658392018061'],
+    }),
+    ('bleach', '3.3.1', {
```
Comment (Micket) [inline easybuild/easyconfigs/s/Spyder/Spyder-5.0.5-foss-2020b.eb:49]: Already included in deps (slightly different version, but maybe still compatible)

File: easybuild/easyconfigs/s/Spyder/Spyder-5.0.5-foss-2020b.eb [hunk 4]
```diff
@@ -0,0 +1,278 @@
...
+        'checksums': ['306483a5a9795474160ad57fce3ddd1b50551e981eed8e15a582d34cef28aafa'],
+    }),
+    ('cloudpickle', '1.6.0', {
+        'checksums': ['9bc994f9e9447593bd0a45371f0e7ac7333710fcf64a4eb9834bf149f4ef2f32'],
+    }),
+    ('cookiecutter', '1.7.3', {
+        'checksums': ['6b9a4d72882e243be077a7397d0f1f76fe66cf3df91f3115dbb5330e214fa457'],
+    }),
+    ('decorator', '5.0.9', {
```
Comment (Micket) [inline easybuild/easyconfigs/s/Spyder/Spyder-5.0.5-foss-2020b.eb:58]: Already included in deps (different version, but maybe still compatible?)

File: easybuild/easyconfigs/s/Spyder/Spyder-5.0.5-foss-2020b.eb [hunk 5]
```diff
@@ -0,0 +1,278 @@
...
+        'checksums': ['07528381786f2a6237b061f6e96610a4167b226cb926e2aa2b6b1d78057c576b'],
+    }),
+    ('helpdev', '0.7.1', {
+        'checksums': ['bb62a79acbac141dadf42cadeb92bb7450dd18b9824a62043b6a0b149190db3d'],
+    }),
+    ('isort', '5.9.2', {
+        'use_pip': False,
+        'checksums': ['f65ce5bd4cbc6abdfbe29afc2f0245538ab358c14590912df638033f157d555e'],
+    }),
+    ('jedi', '0.17.2', {
```
Comment (Micket) [inline easybuild/easyconfigs/s/Spyder/Spyder-5.0.5-foss-2020b.eb:74]: Already included in deps

File: easybuild/easyconfigs/s/Spyder/Spyder-5.0.5-foss-2020b.eb [hunk 6]
```diff
@@ -0,0 +1,278 @@
...
+        'checksums': ['86ed7d9b750603e4ba582ea8edc678657fb4007894a12bcf6f4bb97892f31d20'],
+    }),
+    ('jeepney', '0.7.0', {
```
Comment (Micket) [inline easybuild/easyconfigs/s/Spyder/Spyder-5.0.5-foss-2020b.eb:77]: Already included in deps (different version, but maybe still compatible?)

File: easybuild/easyconfigs/s/Spyder/Spyder-5.0.5-foss-2020b.eb [hunk 7]
```diff
@@ -0,0 +1,278 @@
...
+        'use_pip': False,
+        'checksums': ['1237cd64c8f7ac3aa4b3f332c4d0fb4a8216f39eaa662ec904302d4d77de5a54'],
+    }),
+    ('arrow', '1.1.1', {
+        'checksums': ['dee7602f6c60e3ec510095b5e301441bc56288cb8f51def14dcb3079f623823a'],
+    }),
+    ('jinja2-time', '0.2.0', {
+        'checksums': ['d14eaa4d315e7688daa4969f616f226614350c48730bfa1692d2caebd8c90d40'],
+    }),
+    ('jupyterlab_pygments', '0.1.2', {
```
Comment (Micket) [inline easybuild/easyconfigs/s/Spyder/Spyder-5.0.5-foss-2020b.eb:87]: Already included in deps

File: easybuild/easyconfigs/s/Spyder/Spyder-5.0.5-foss-2020b.eb [hunk 8]
```diff
@@ -0,0 +1,278 @@
...
+        'checksums': ['cfcda0873626150932f438eccf0f8bf22bfa92345b814890ab360d666b254146'],
+    }),
+    ('importlib_metadata', '4.6.1', {
+        'checksums': ['079ada16b7fc30dfbb5d13399a5113110dab1aa7c2bc62f66af75f0b717c8cac'],
+    }),
+    ('keyring', '21.8.0', {
```
Comment (Micket) [inline easybuild/easyconfigs/s/Spyder/Spyder-5.0.5-foss-2020b.eb:93]: Already included in deps (slightly different version, but maybe still compatible?)

File: easybuild/easyconfigs/s/Spyder/Spyder-5.0.5-foss-2020b.eb [hunk 9]
```diff
@@ -0,0 +1,278 @@
...
+        'preinstallopts': """sed -i 's/setup()/setup(version="%(version)s")/g' setup.py && """,
+        'checksums': ['1746d3ac913d449a090caf11e9e4af00e26c3f7f7e81027872192b2398b98675'],
+    }),
+    ('keyrings.alt', '4.1.0', {
```
Comment (Micket) [inline easybuild/easyconfigs/s/Spyder/Spyder-5.0.5-foss-2020b.eb:97]: Already included in deps (slightly different version, but maybe still compatible?)

File: easybuild/easyconfigs/s/Spyder/Spyder-5.0.5-foss-2020b.eb [hunk 10]
```diff
@@ -0,0 +1,278 @@
...
+        'preinstallopts': """sed -i 's/setup()/setup(version="%(version)s")/g' setup.py && """,
+        'checksums': ['52ccb61d6f16c10f32f30d38cceef7811ed48e086d73e3bae86f0854352c4ab2'],
+    }),
+    ('lazy-object-proxy', '1.6.0', {
+        'checksums': ['489000d368377571c6f982fba6497f2aa13c6d1facc40660963da62f5c379726'],
+    }),
+    ('mccabe', '0.6.1', {
+        'checksums': ['dd8d182285a0fe56bace7f45b5e7d1a6ebcbf524e8f3bd87eb0f125271b8831f'],
+    }),
+    ('jupyter_client', '6.1.12', {
```
Comment (Micket) [inline easybuild/easyconfigs/s/Spyder/Spyder-5.0.5-foss-2020b.eb:107]: Already included in deps (slightly different version, but maybe still compatible)

File: easybuild/easyconfigs/s/Spyder/Spyder-5.0.5-foss-2020b.eb [hunk 11]
```diff
@@ -0,0 +1,278 @@
...
+        'checksums': ['c4bca1d0846186ca8be97f4d2fa6d2bae889cce4892a167ffa1ba6bd1f73e782'],
+    }),
+    ('nbclient', '0.5.3', {
```
Comment (Micket) [inline easybuild/easyconfigs/s/Spyder/Spyder-5.0.5-foss-2020b.eb:110]: Already included in deps (slightly different version, but maybe still compatible)

File: easybuild/easyconfigs/s/Spyder/Spyder-5.0.5-foss-2020b.eb [hunk 12]
```diff
@@ -0,0 +1,278 @@
...
+        'checksums': ['db17271330c68c8c88d46d72349e24c147bb6f34ec82d8481a8f025c4d26589c'],
+    }),
+    ('nest_asyncio', '1.5.1', {
```
Comment (Micket) [inline easybuild/easyconfigs/s/Spyder/Spyder-5.0.5-foss-2020b.eb:113]: Already included in deps (slightly different version, but maybe still compatible)

File: easybuild/easyconfigs/s/Spyder/Spyder-5.0.5-foss-2020b.eb [hunk 13]
```diff
@@ -0,0 +1,278 @@
...
+    ('pycodestyle', '2.7.0', {
+        'checksums': ['c389c1d06bf7904078ca03399a4816f974a1d590090fecea0c63ec26ebaf1cef'],
+    }),
+    ('pydocstyle', '6.1.1', {
+        'checksums': ['1d41b7c459ba0ee6c345f2eb9ae827cab14a7533a88c5c6f7e94923f72df92dc'],
+    }),
+    ('pyflakes', '2.3.1', {
+        'checksums': ['f5bc8ecabc05bb9d291eb5203d6810b49040f6ff446a756326104746cc00c1db'],
+    }),
+    ('isort', '5.9.2', {
```
Comment (Micket) [inline easybuild/easyconfigs/s/Spyder/Spyder-5.0.5-foss-2020b.eb:134]: Duplicate, one with and one without use_pip?

File: easybuild/easyconfigs/s/Spyder/Spyder-5.0.5-foss-2020b.eb [hunk 14]
```diff
@@ -0,0 +1,278 @@
...
+        'checksums': ['f65ce5bd4cbc6abdfbe29afc2f0245538ab358c14590912df638033f157d555e'],
+    }),
+    ('pylint', '2.9.5', {
+        'checksums': ['1f333dc72ef7f5ea166b3230936ebcfb1f3b722e76c980cb9fe6b9f95e8d3172'],
+    }),
+    ('ujson', '4.0.2', {
```
Comment (Micket) [inline easybuild/easyconfigs/s/Spyder/Spyder-5.0.5-foss-2020b.eb:140]: Already included in deps (slightly different version, but maybe still compatible)

File: easybuild/easyconfigs/s/Spyder/Spyder-5.0.5-foss-2020b.eb [hunk 15]
```diff
@@ -0,0 +1,278 @@
...
+        'checksums': ['c615a9e9e378a7383b756b7e7a73c38b22aeb8967a8bfbffd4741f7ffd043c4d'],
+    }),
+    ('python-language-server', '0.36.2', {
+        'modulename': 'pyls',
+        'checksums': ['9984c84a67ee2c5102c8e703215f407fcfa5e62b0ae86c9572d0ada8c4b417b0'],
+    }),
+    ('pyrsistent', '0.18.0', {
```
Comment (Micket) [inline easybuild/easyconfigs/s/Spyder/Spyder-5.0.5-foss-2020b.eb:147]: Already included in deps (slightly different version, but maybe still compatible)

File: easybuild/easyconfigs/s/Spyder/Spyder-5.0.5-foss-2020b.eb [hunk 16]
```diff
@@ -0,0 +1,278 @@
...
+    ('QtAwesome', '1.0.3', {
+        'checksums': ['d37bbeb69ddc591e5ff036b741bda8d1d92133811f1f5a7150021506f70b8e6e'],
+    }),
+    ('qtconsole', '5.1.1', {
+        'checksums': ['bbc34bca14f65535afcb401bc74b752bac955e5313001ba640383f7e5857dc49'],
+    }),
+    ('QtPy', '1.9.0', {
+        'checksums': ['2db72c44b55d0fe1407be8fba35c838ad0d6d3bb81f23007886dc1fc0f459c8d'],
+    }),
+    ('SecretStorage', '3.3.1', {
```
Comment (Micket) [inline easybuild/easyconfigs/s/Spyder/Spyder-5.0.5-foss-2020b.eb:177]: Already included in deps (vera different version, but maybe still compatible?)

File: easybuild/easyconfigs/s/Spyder/Spyder-5.0.5-foss-2020b.eb [hunk 17]
```diff
@@ -0,0 +1,278 @@
...
+        'modulename': 'secretstorage',
+        'checksums': ['fd666c51a6bf200643495a04abb261f83229dcb6fd8472ec393df7ffc8b6f195'],
+    }),
+    ('three-merge', '0.1.1', {
+        'checksums': ['60f6afe144595560d63ae32625351bcef3b94733b54eb97800a9feb0f3d9d970'],
+    }),
+    ('parso', '0.7.0', {
```
Comment (Micket) [inline easybuild/easyconfigs/s/Spyder/Spyder-5.0.5-foss-2020b.eb:184]: Already included in deps (different version, but maybe still compatible)

File: easybuild/easyconfigs/s/Spyder/Spyder-5.0.5-foss-2020b.eb [hunk 18]
```diff
@@ -0,0 +1,278 @@
...
+        'checksums': ['b1b5c06b5cd9902f9746a083fd760bb4de3fa4129bc8c1c1ef2bcd84002bc45f'],
+    }),
+    ('spyder-kernels', '2.0.5', {
+        'modulename': 'spyder_kernels',
+        'checksums': ['4428a9aa8f94987f366fad3afda427933a131cca5fd9c361c8553c564eae011e'],
+    }),
+    ('text-unidecode', '1.3', {
+        'checksums': ['bad6603bb14d279193107714b288be206cac565dfa49aa5b105294dd5c4aab93'],
+    }),
+    ('traitlets', version, {
```
Comment (Micket) [inline easybuild/easyconfigs/s/Spyder/Spyder-5.0.5-foss-2020b.eb:201]: Already included in deps (slightly different version, but maybe still compatible). I also think the "version" here would be wrong regardless

File: easybuild/easyconfigs/s/Spyder/Spyder-5.0.5-foss-2020b.eb [hunk 19]
```diff
@@ -0,0 +1,278 @@
...
+        'checksums': ['50b6f157849174217d0656f99dc82fe932884fb250826c18350e159ec6cdf342'],
+    }),
+    ('pathspec', '0.9.0', {
+        'checksums': ['e564499435a2673d586f6b2130bb5b95f04a3ba06f81b8f895b651a3c76aabb1'],
+    }),
+    ('mypy-extensions', '0.4.3', {
+        'sources': ['mypy_extensions-%(version)s.tar.gz'],
+        'checksums': ['2d82818f5bb3e369420cb3c4060a7970edba416647068eb4c5343488a6c604a8'],
+    }),
+    ('flit_core', '3.2.0', {
```
Comment (Micket) [inline easybuild/easyconfigs/s/Spyder/Spyder-5.0.5-foss-2020b.eb:230]: Already included in deps (different version, but maybe still compatible)

**Review by Micket — CHANGES_REQUESTED** [review]
I think it's always best to keep using the existing package versions if they exist for maximum compatibility with all other stuff, which might rely on those versions

**Comment by boegel** [comment]
closing this since `foss/2020b` is deprecated, see https://docs.easybuild.io/policies/toolchains

**Pull request closed without merging**